File: src/main/gamma-helper.swift
import CoreGraphics
import Foundation

let args = CommandLine.arguments

if args.count < 4 {
  exit(1)
}

guard let redMax = Float(args[1]),
      let greenMax = Float(args[2]),
      let blueMax = Float(args[3]) else {
  exit(1)
}

var displayCount: UInt32 = 0
CGGetActiveDisplayList(0, nil, &displayCount)
var displays = [CGDirectDisplayID](repeating: 0, count: Int(displayCount))
CGGetActiveDisplayList(displayCount, &displays, &displayCount)

for display in displays {
  CGSetDisplayTransferByFormula(
    display,
    0, 1, redMax,
    0, 1, greenMax,
    0, 1, blueMax
  )
}

File: src/main/gamma-helper.c
#include <CoreGraphics/CoreGraphics.h>
#include <stdlib.h>

int main(int argc, char *argv[]) {
    if (argc < 4) return 1;

    float red   = (float)atof(argv[1]);
    float green = (float)atof(argv[2]);
    float blue  = (float)atof(argv[3]);

    uint32_t displayCount = 0;
    CGGetActiveDisplayList(0, NULL, &displayCount);
    if (displayCount == 0) return 0;

    CGDirectDisplayID displays[16];
    CGGetActiveDisplayList(displayCount, displays, &displayCount);

    for (uint32_t i = 0; i < displayCount; i++) {
        CGSetDisplayTransferByFormula(displays[i],
            0, red,   1.0,
            0, green, 1.0,
            0, blue,  1.0);
    }
    return 0;
}
